src/domain/SymbolTable.java

```diff
@@ -17,6 +20,14 @@
         public HashNode(K key) {
             this.key = key;
         }
+
+        @Override
+        public String toString() {
+            return "HashNode{" +
+                    "key=" + key +
+                    ", next=" + next +
+                    '}';
+        }
```

Comment:
This shows nulls too in the SymbolTable.in, and it is very confusing why you have nulls.
If you need to do a garbage colector for example, you would have unnecessary operations if you read the sym table from the file.